pkg/stacks/unpack.go

```diff
@@ -218,13 +238,97 @@ func (sp *StackPackage) IsNamespaced() bool {
 
 // SetApp sets the Stack's App metadata
 func (sp *StackPackage) SetApp(app v1alpha1.AppMetadataSpec) {
-	sp.Stack.Spec.AppMetadataSpec = app
+	app.DeepCopyInto(&sp.Stack.Spec.AppMetadataSpec)
+
 	sp.appSet = true
 }
 
-// SetBehavior sets the Stack's Behavior configuration
-func (sp *StackPackage) SetBehavior(behavior unstructured.Unstructured) {
-	sp.Behavior = behavior
+// SetStackConfig sets the Stack's Behavior using StackConfig
+func (sp *StackPackage) SetStackConfig(config unstructured.Unstructured) {
+	sp.StackConfig = config
+}
+
+func (sp *StackPackage) createBehaviorController(sd v1alpha1.StackDefinitionBehavior) appsv1.DeploymentSpec {
+	// TODO(displague) where does the name come from? behavior? cmd line arg?
+	// sp.Stack.GetName() is not available when this runs
+	controllerDeploymentName := "tc-" + "something"
+	controllerContainerName := controllerDeploymentName
+	controllerImageName := "bymafmaf/templating-controller:0.0.1"
+
+	// TODO(displague) Where do these files reside within the Stack?
+	// are sp.baseDir or sd.Source.Path useful here?
+	// are we always mounting a remote source image?
+	behaviorDirName := "/"
+	behaviorContentsVolumeName := "behaviors"
+	behaviorMountPoint := "/behaviors"
+	spec := appsv1.DeploymentSpec{
+		Selector: &metav1.LabelSelector{MatchLabels: map[string]string{"app": controllerDeploymentName}},
+		Template: corev1.PodTemplateSpec{
+			ObjectMeta: metav1.ObjectMeta{
+				Labels: map[string]string{"app": controllerDeploymentName},
+			},
+			Spec: corev1.PodSpec{
+				RestartPolicy: corev1.RestartPolicyNever,
+				InitContainers: []corev1.Container{
+					{
+						Name:    "stack-package",
+						Image:   sd.Source.Image,
+						Command: []string{"cp", "-xR", behaviorDirName, behaviorMountPoint},
+						VolumeMounts: []corev1.VolumeMount{
+							{
+								Name:      behaviorContentsVolumeName,
+								MountPath: behaviorMountPoint,
+							},
+						},
+					},
+				},
+				Containers: []corev1.Container{
+					{
+						Name:  controllerContainerName,
+						Image: controllerImageName,
+						Args: []string{
+							"manager",
+							"--resources-dir", behaviorMountPoint,
+							"--stack-definition-namespace", sp.Stack.GetNamespace(),
```

Comment:
These flags are not filled. Looking at your comment above, it seems because `sp.Stack` is empty at this step.

Reply:
`unpack` does not know the name of the StackInstall / Stack / etc.  This is why the StackManager currently applies the name.  This is only becoming a problem in my PR because the name has to be injected in a slice of command line arguments instead of in a well known location (deployment name, template name, stack name, etc).

`unpack.go:NewStackPackage(baseDir string)` could take a name parameter, which it could acquire through `--stack-name` and `--stack-namespace` arguments to `unpack` (set via environment, or we could use a configmap).

Reply:
> Then the name of the Stack/StackDefinition doesn't matter, and it can be generated by the same code which is generating the StackDefinition

A drawback of this is that the StackInstall and Stack names would not match, but in considering that StackInstalls could in the future lead to multiple Stack records, the names wouldn't have the luxury of such simple naming for ever. I defer to https://github.com/crossplaneio/crossplane/issues/1089#issuecomment-571814987 for thoughts how to approach that problem.